src/main/java/com/google/sps/algorithms/AbuseDetection.java

```diff
@@ -0,0 +1,95 @@
+package com.google.sps.algorithms;
+
+import com.google.sps.data.MatchRepository;
+import com.google.sps.data.MatchRequest;
+import com.google.sps.data.PersistentUserRepository;
+import com.google.sps.data.User;
+import com.google.sps.data.UserRepository;
+import java.lang.Exception;
+import java.io.IOException; 
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+
+
+public final class AbuseDetection {
+
+    // private Collection <LocalTime> timesOfRequests = new ArrayList<LocalTime>();
+    private Collection <Date> timesOfRequests = new ArrayList<Date>();
+
+    private int requestCounter = 0;
+    private int requestsDropped = 0;
+
+    private final int currentNumRequestsAllowed;
+    private final long timePeriod;
+  
+  /**
+  * Constructor that initializes currentNumRequestsAllowed and timePeriod
+  * The timeValue represents the amount of seconds allowed for this interval.
+  * The requestsAllowed is the amount of requests that are being allowed
+  * during that timeValue interval.
+  */
+  public AbuseDetection(long timeValue, int requestsAllowed) { 
```

Comment:
maybe call this durationInSeconds so that readers understand what units to use.
Better yet, you can pass a Duration object here so make the meaning unambiguous.

Reply:
Ok. Why pass a Duration Object.

Reply:
Did this.